js/todoOperations.js

```diff
@@ -0,0 +1,128 @@
+function addTaskToTable(task, todos) {
+    const tableBody = document.querySelector('.todo-table__body');
+    const row = document.createElement('tr');
+    row.classList.add('todo-table__row');
+
+    if (task.completed) {
+        row.classList.add('completed');
+    }
+
+    row.appendChild(createCell(task.id, 'todo-table__cell'));
+    row.appendChild(createCell(task.todo, 'todo-table__cell'));
+    row.appendChild(createCell(task.userId, 'todo-table__cell'));
+    row.appendChild(createCell(task.completed ? 'Completed' : 'Pending', 'todo-table__cell'));
+
+    const buttonsCell = createCell('', 'todo-table__cell');
+    const buttonsContainer = document.createElement('div');
+    buttonsContainer.classList.add('todo-table__cell--buttons');
+
+    const deleteButton = createButton('Delete', 'todo-table__button todo-table__button--delete', () => handleDelete(row, task.id));
+    const doneButton = createButton(task.completed ? 'Undo' : 'Done', 'todo-table__button todo-table__button--done', () => handleDone(row, task, doneButton, todos));
+
+    buttonsContainer.append(deleteButton, doneButton);
+    buttonsCell.appendChild(buttonsContainer);
+    row.appendChild(buttonsCell);
+    tableBody.appendChild(row);
+
+    row.children[1].addEventListener('click', () => handleEdit(row, task));
+    updateTasksCount(tableBody.children.length);
+}
+
+function handleSearch() {
```

Comment:
handleSearch in the first time appears like an action to be called every time the user search for a new todos, so let's have a better naming, for example `initializeSearchListener`